### PR TITLE
BaOverlay: resolve measurement overlay

### LINKED ISSUE #2265: Consolidate BA overlay classes
BaOverlay and MeasurementOverlay needs to be refactored:

- switching to MvuElement
- simplified to a single component

### PR DESCRIPTION
**Pull request by thiloSchlemmer**
resolves #2265

### CODE REVIEW

File: src/modules/olMap/components/BaOverlay.js [hunk 1]
```diff
@@ -63,83 +116,101 @@ export class BaOverlay extends BaElement {
 		`;
 	}
 
-	/**
-	 * @protected
-	 */
-	_updatePosition() {
-		this._position = this.geometry.getLastCoordinate();
-	}
+	_getContent(model) {
```
Comment (taulinger) [inline src/modules/olMap/components/BaOverlay.js:119]: could be a private method

File: src/modules/olMap/components/BaOverlay.js [hunk 2]
```diff
@@ -63,83 +116,101 @@ export class BaOverlay extends BaElement {
...
 
 	set isDraggable(value) {
-		if (value !== this.isDraggable) {
-			this._isDraggable = value;
-			this.render();
-		}
+		this.signal(Update_Draggable, value);
 	}
 
 	get isDraggable() {
```
Comment (taulinger) [inline src/modules/olMap/components/BaOverlay.js:193]: Not covered by a test

File: src/modules/olMap/components/BaOverlay.js [hunk 3]
```diff
@@ -63,83 +116,101 @@ export class BaOverlay extends BaElement {
...
+				return value;
+		}
 	}
 
 	static get tag() {
 		return 'ba-map-overlay';
 	}
 
+	_getMeasuredLength = (geometry) => {
+		const alreadyMeasuredLength = geometry ? geometry.get(PROJECTED_LENGTH_GEOMETRY_PROPERTY) : null;
```
Comment (taulinger) [inline src/modules/olMap/components/BaOverlay.js:161]: `null` branch not covered by a test

File: src/modules/olMap/components/BaOverlay.js [hunk 4]
```diff
@@ -3,58 +3,111 @@
...
-		this._isDraggable = false;
+		super({
+			value: null,
+			floating: true,
+			overlayType: BaOverlayTypes.TEXT,
+			draggable: false,
+			placement: Default_Placement,
+			geometry: null,
+			position: null
+		});
```
Comment (taulinger) [inline src/modules/olMap/components/BaOverlay.js:56]: no test for default/initial model

File: src/modules/olMap/overlayStyle/MeasurementOverlayStyle.js
Comment (taulinger) [inline src/modules/olMap/overlayStyle/MeasurementOverlayStyle.js:1]: Some branches are not covered here, can we improve that without great expense?

**Review by taulinger — COMMENTED** [review]
Nice!

File: src/modules/olMap/components/BaOverlay.js (hunk 3 above)
Comment (thiloSchlemmer) [reply src/modules/olMap/components/BaOverlay.js:161]: Removing null-check. In all usecases the geometry-object is already checked.

File: test/modules/olMap/overlayStyle/MeasurementOverlayStyle.test.js [hunk 5]
```diff
@@ -987,4 +1205,15 @@ describe('MeasurementOverlayStyle', () => {
...
+		it('finds a placement sector for all degrees within 360°', () => {
+			// regression test -> to be safe against changes in private SectorsOfPlacement definitions
+			const classUnderTest = new MeasurementOverlayStyle();
+
+			for (let angle = 0; angle <= 360; angle++) {
+				expect(classUnderTest._getPlacement(angle).sector).not.toBeUndefined();
+			}
+		});
+	});
 });
```
Comment (taulinger) [inline test/modules/olMap/overlayStyle/MeasurementOverlayStyle.test.js:1219]: The test won't run isolated

```
 Error: No registered instance or factory found for "MapService" in /tmp/_karma_webpack_480479/commons.js (line 1330)
      ./src/injection/core/injector.js/inject/<@/tmp/_karma_webpack_480479/commons.js:1330:15
```

**Review by taulinger — APPROVED** [review]
THX

**Pull request merged**